jetty-core/jetty-server/src/main/java/org/eclipse/jetty/server/handler/ContextHandler.java

```diff
@@ -733,11 +733,23 @@ protected ClassLoader enterScope(Request contextRequest)
         ClassLoader lastLoader = Thread.currentThread().getContextClassLoader();
         __context.set(_context);
         if (_classLoader != null)
-            Thread.currentThread().setContextClassLoader(_classLoader);
+            setContextClassLoader(_classLoader);
         notifyEnterScope(contextRequest);
         return lastLoader;
     }
 
+    private void setContextClassLoader(ClassLoader classLoader)
+    {
+        try
+        {
+            Thread.currentThread().setContextClassLoader(classLoader);
+        }
+        catch (SecurityException e)
```

Comment:
Do the null check in the new method only.
That way you catch it for both enter & exit

```suggestion
    private void setContextClassLoader(ClassLoader classLoader)
    {
        if (classLoader == null)
            return;
        try
        {
            Thread.currentThread().setContextClassLoader(classLoader);
        }
        catch (SecurityException e)
```

Reply:
I don't think we can skip the call if `classLoader==null`.

Reply:
Grrr...

The call `Thread.setContextClassaLoader(null)` has meaning, it's not just a no-op.

* https://docs.oracle.com/en/java/javase/17/docs/api/java.base/java/lang/Thread.html#setContextClassLoader(java.lang.ClassLoader)

> **Parameters**:
> `cl` - the context ClassLoader for this Thread, or null indicating the system class loader (or, failing that, the bootstrap class loader)

Reply:
I think we can skip the call.   If we say that a `ContextHandler` with a null ClassLoader has no requirement for its TCCL, then it is fine to skip the call.   If a context specifically needs the system or bootstrap class loader, then they can be explicitly set on the `ContextHandler`.

This is a slight change of meaning/behaviour - we make a distinction between a context that doesn't care about the classloader vs one that wants a specific one.

Reply:
This is a potentially breaking change, but I don't think anyone would be impacted by this new behavior which I think is saner.

Let me do this.